### ISSUE TITLE
Circular properties are not supported

### ISSUE THREAD

**Issue by jonaslagoni** [jonaslagoni/asyncapi-quicktype-filter#15]
While fixing the following issue https://github.com/jonaslagoni/asyncapi-quicktype-template/issues/4 I noticed that circular dependencies where not supported in the filter.

The line that causes problems are:
https://github.com/jonaslagoni/asyncapi-quicktype-filter/blob/11232c998446098d23c6755e130b62840da6c4bb/src/index.js#L55
because JSON.stringify does not support this. Maybe check out [existing implementations](https://github.com/quicktype/quicktype-vscode/blob/master/src/extension.ts) how they work

**Comment by jonaslagoni** [comment]
Because of how the parser work it is not possible to get a raw JSON schema for a specific AsyncAPI payload with definitions and refs. 

Possible solutions:
Build the JSON Schema from the payload with refs and definitions so it is complete when given to QuickType.

**Comment by mmacai** [comment]
Hi, any update on this?

**Comment by jonaslagoni** [comment]
Hey @mmacai, it would probably be possible to fix this for QuickType, however, our own [replacement](https://github.com/asyncapi/generator-model-sdk) is almost done where I will focus on integrating that library into a template for easy usage. If you really need this for the QuickType filter I would accept any PRs fixing this 😄

**Comment by mmacai** [comment]
Hi @jonaslagoni, pretty much just looking how should I generate typescript types from the asyncapi yaml file. With this one it ignores `refs`, so I used openapi generator in a bit hacky way. Would this work then https://github.com/asyncapi/generator-model-sdk?

**Comment by jonaslagoni** [comment]
@mmacai yes TypeScript is one of the initially supported languages and you can provide it with your entire AsyncAPI file and it would render all the models for message payloads. Just keep in mind we don't have a wrapper/cli for the tool yet, so you would have to provide your own functionality to write the models to a file. Feel free to create any issues/feature requests if you try it out - https://github.com/asyncapi/generator-model-sdk/issues
